Bump the bytecode reading version kMaxSupportedBytecodeVersion to 6

iseeyuan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#59714 Bump the bytecode reading version kMaxSupportedBytecodeVersion to 6**
Bytecode v6 is on implicit operator versioning through number of specified arguments. Both the read and write codes are available. This PR is to enable reading v6 models. The default writing format is not changed yet and will be bumped in a later PR.

Test: CI.
Local: change the writing version to 6 temporally and run the unit tests in LiteInterpreterTest. There are a number of end-to-end tests to write v6 bytecode, read and run it.

Differential Revision: [D29007538](https://our.internmc.facebook.com/intern/diff/D29007538)

iseeyuan committed:
Bump the bytecode reading version kMaxSupportedBytecodeVersion to 6

[ghstack-poisoned]

raziel commented on caffe2/serialize/versions.h:
```diff
@@ -78,7 +78,7 @@ static_assert(kProducedBytecodeVersion >= kProducedFileFormatVersion,
 // handle an updated operator.
 constexpr uint64_t kMinSupportedBytecodeVersion = 0x3L;
-constexpr uint64_t kMaxSupportedBytecodeVersion = 0x5L;
+constexpr uint64_t kMaxSupportedBytecodeVersion = 0x6L;
```
we should then probably update also the docs above with each version (currently this only covers v4).

iseeyuan committed:
Update on "Bump the bytecode reading version kMaxSupportedBytecodeVersion to 6"

Bytecode v6 is on implicit operator versioning through number of specified arguments. Both the read and write codes are available. This PR is to enable reading v6 models. The default writing format is not changed yet and will be bumped in a later PR.

Test: CI.
Local: change the writing version to 6 temporally and run the unit tests in LiteInterpreterTest. There are a number of end-to-end tests to write v6 bytecode, read and run it.

[ghstack-poisoned]

iseeyuan commented:
@iseeyuan has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29007538).

lutzroeder commented:
@iseeyuan @Raziel `version=6` is not supported in PyTorch 1.9 -- right? Is this planned for PyTorch 1.10 or 2.0?
